Move _test_inductor_realize into python

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #90517

Addresses https://github.com/pytorch/pytorch/pull/90014/files#r1043625932

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

peterbell10 committed:
Move _test_inductor_realize into python

Addresses https://github.com/pytorch/pytorch/pull/90014/files#r1043625932

[ghstack-poisoned]

peterbell10 committed:
Update on "Move _test_inductor_realize into python"


Addresses https://github.com/pytorch/pytorch/pull/90014/files#r1043625932

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

peterbell10 committed:
Update on "Move _test_inductor_realize into python"


Addresses https://github.com/pytorch/pytorch/pull/90014/files#r1043625932

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

ngimel reviewed (approved):
Pretty cool that everything can be done in python now

peterbell10 commented:
@pytorchbot merge

peterbell10 commented:
@pytorchbot merge

albanD commented on torch/_inductor/test_operators.py:
```diff
@@ -0,0 +1,24 @@
+_test_lib_def.define("realize(Tensor self) -> Tensor")
+
+_test_lib_impl = torch.library.Library("_inductor_test", "IMPL")
+for dispatch_key in ("CPU", "CUDA", "Meta"):
```
Given that you are doing an autograd-aware implementation, you should register this to `AutogradCPU`, `AutogradCUDA` and `AutogradMeta`.

albanD replied:
My bad I misread this. I though the custom Function was registered on these keys.

peterbell10 committed:
Fix merge conflict on "Move _test_inductor_realize into python"


Addresses https://github.com/pytorch/pytorch/pull/90014/files#r1043625932

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

peterbell10 commented:
@pytorchbot merge
